lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -7373,6 +7373,13 @@ namespace Js
         ::Math::RecordOverflowPolicy newLenOverflow;
         uint32 newLen = UInt32Math::Add(len - deleteLen, insertLen, newLenOverflow); // new length of the array after splice
 
+        // If newLen overflowed, take the slower path to do splicing.
+        if (newLenOverflow.HasOverflowed())
+        {
+            pArr = EnsureNonNativeArray(pArr);
```

Comment:
>pArr = EnsureNonNativeArray(pArr); [](start = 12, length = 34)

nit: should we have if (insertArgs) check, though doesn't matter much for such overflow cases.

Reply:
It shouldn't matter with overflow case.. as the TruncateToProperties can also go to the user code.